src/GeneralStateTestsFiller/stTransactionTest/CreateTransactionEOF1Filler.yml

```diff
@@ -30,34 +30,121 @@ CreateTransactionEOF1:
       #- ':yul { mstore8(0, 0xef) return(0, 32) }'
       - ':label deploying_0xef00..00 :raw 0x60ef60005360206000f3'
 
-
-      # Valid EOF1 initcode and code combinations (both with and withoug data section)
-
-      # Legacy initcode => EOF code
+     # Legacy initcode => EOF code
       # deployed code without data section
       # deployed code section: 'PUSH1 0 STOP'
       - ':label deploying_valid_eof1 :yul { mstore(0, 0xef00010100030060000000000000000000000000000000000000000000000000) return(0, 10) }'
+      # deployed code section: 'PUSH1 0 PUSH1 0 RETURN'
+      - ':label deploying_valid_eof1_return :yul { mstore(0, 0xef00010100050060006000f30000000000000000000000000000000000000000) return (0, 12) }'
+      # deployed code section: 'PUSH1 1 PUSH 0 SSTORE PUSH1 00 REVERT'
+      - ':label deploying_valid_eof1_revert :yul { mstore(0, 0xef000101000a00600160005560006000fd000000000000000000000000000000)  return(0, 17) }'
+      # deployed code section: 'PUSH1 0 INVALID'
+      - ':label deploying_valid_eof1_invalid :yul { mstore(0, 0xef0001010003006000fe00000000000000000000000000000000000000000000) return (0, 10) }'
+      # deployed code section: 'PUSH1 0 SELFDESTRUCT'
+      - ':label deploying_valid_eof1_selfdestruct :yul { mstore(0, 0xef0001010003006000ff00000000000000000000000000000000000000000000) return (0, 10) }'
       # deployed code with data section
       # deployed code section: 'PUSH1 0 STOP' data section: 'aabbccdd'
       - ':label deploying_valid_eof1_with_data :yul { mstore(0, 0xef000101000302000400600000aabbccdd000000000000000000000000000000) return(0, 17) }'
+      # deployed code section: 'PUSH1 0 PUSH1 0 RETURN' data section: 'aabbccdd'
```

Comment:
I would maybe remove new cases that check the same validation for EOF with data section.

Data section should not affect anything about validating code section, so I'd just leave the cases with "no data section" variant.